zeus/device/gpu.py

```diff
@@ -0,0 +1,829 @@
+        """Reset the memory locked clocks to default values."""
+        pass
+
+    @abc.abstractmethod
+    def resetGpuLockedClocks(self) -> None:
+        """Reset the GPU locked clocks to default values."""
+        pass
+
+    @abc.abstractmethod
+    def getPowerUsage(self) -> int:
+        """Return the current power usage of the GPU. Units: mW."""
+        pass
+
+    @abc.abstractmethod
+    def supportsGetTotalEnergyConsumption(self) -> bool:
+        """Check if the GPU supports retrieving total energy consumption."""
+        pass
+
+    @abc.abstractmethod
+    def getTotalEnergyConsumption(self) -> int:
+        """Return the total energy consumption of the GPU since driver load. Units: mJ."""
+        pass
+
+
+""" GPU MANAGEMENT OBJECTS """
+
+
+class NVIDIAGPU(GPU):
+    """Control a Single NVIDIA GPU.
+
+    Uses NVML Library to control and query GPU. There is a 1:1 mapping between the methods in this class and the NVML library functions.
+    Zeus GPU Exceptions are raised when NVML errors occur.
+    To ensure computational efficiency, this class utilizes caching (ex. saves the handle) to avoid repeated calls to NVML.
+    """
+
+    def __init__(self, gpu_index: int) -> None:
+        """Initializes the NVIDIAGPU object with a specified GPU index. Acquires a handle to the GPU using `pynvml.nvmlDeviceGetHandleByIndex`."""
+        super().__init__(gpu_index)
+        self._get_handle()
+        self._supportsGetTotalEnergyConsumption = None
+
+    _exception_map = {
+        pynvml.NVML_ERROR_UNINITIALIZED: ZeusGPUInitError,
+        pynvml.NVML_ERROR_INVALID_ARGUMENT: ZeusGPUInvalidArgError,
+        pynvml.NVML_ERROR_NOT_SUPPORTED: ZeusGPUNotSupportedError,
+        pynvml.NVML_ERROR_NO_PERMISSION: ZeusGPUNoPermissionError,
+        pynvml.NVML_ERROR_ALREADY_INITIALIZED: ZeusGPUAlreadyInitializedError,
+        pynvml.NVML_ERROR_NOT_FOUND: ZeusGPUNotFoundError,
+        pynvml.NVML_ERROR_INSUFFICIENT_SIZE: ZeusGPUInsufficientSizeError,
+        pynvml.NVML_ERROR_INSUFFICIENT_POWER: ZeusGPUInsufficientPowerError,
+        pynvml.NVML_ERROR_DRIVER_NOT_LOADED: ZeusGPUDriverError,  # change to : ZeusGPUDriverNotLoadedError
+        pynvml.NVML_ERROR_TIMEOUT: ZeusGPUTimeoutError,
+        pynvml.NVML_ERROR_IRQ_ISSUE: ZeusGPUIRQError,
+        pynvml.NVML_ERROR_LIBRARY_NOT_FOUND: ZeusGPULibraryNotFoundError,
+        pynvml.NVML_ERROR_FUNCTION_NOT_FOUND: ZeusGPUFunctionNotFoundError,
+        pynvml.NVML_ERROR_CORRUPTED_INFOROM: ZeusGPUCorruptedInfoROMError,
+        pynvml.NVML_ERROR_GPU_IS_LOST: ZeusGPULostError,
+        pynvml.NVML_ERROR_RESET_REQUIRED: ZeusGPUResetRequiredError,
+        pynvml.NVML_ERROR_OPERATING_SYSTEM: ZeusGPUOperatingSystemError,
+        pynvml.NVML_ERROR_LIB_RM_VERSION_MISMATCH: ZeusGPULibRMVersionMismatchError,
+        pynvml.NVML_ERROR_MEMORY: ZeusGPUMemoryError,
+        pynvml.NVML_ERROR_UNKNOWN: ZeusGPUUnknownError,
+    }
+
+    @staticmethod
+    def _handle_nvml_errors(func):
```

Comment:
Try removing `staticmethod` and moving this outside of the class. The test failure has to do something about this. Otherwise, I don't know. Either your local environment is not representative (e.g., like last time when you had PyTorch installed and tests passed locally) or testing is not good, you need to investigate.

Reply:
that fixed it, all passing now